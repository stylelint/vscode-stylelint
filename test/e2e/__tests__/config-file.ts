import { openDocument, waitForDiagnostics, assertDiagnostics, closeAllEditors } from '../helpers';

describe('"stylelint.configFile" setting', () => {
	afterEach(async () => {
		await closeAllEditors();
	});

	it('should resolve the config file using the specified path', async () => {
		const document = await openDocument('config/config-file.css');
		const diagnostics = await waitForDiagnostics(document);

		assertDiagnostics(diagnostics, [
			{
				code: 'block-no-empty',
				codeDescription: 'https://stylelint.io/user-guide/rules/block-no-empty',
				message: 'Unexpected empty block (block-no-empty)',
				range: [0, 2, 0, 4],
				severity: 'error',
			},
		]);
	});
});
