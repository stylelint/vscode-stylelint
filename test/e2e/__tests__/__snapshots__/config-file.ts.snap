// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"stylelint.configFile" setting should resolve the config file using the specified path 1`] = `
[
  {
    "code": "block-no-empty",
    "codeDescription": {
      "href": "https://stylelint.io/user-guide/rules/block-no-empty",
    },
    "message": "Unexpected empty block (block-no-empty)",
    "range": {
      "end": {
        "character": 4,
        "line": 0,
      },
      "start": {
        "character": 2,
        "line": 0,
      },
    },
    "severity": 1,
    "source": "Stylelint",
  },
]
`;
